tests/e2e/test_runtime.py

```diff
@@ -224,3 +224,62 @@ def test_terminate_process_sigint():
             process.kill()
 
     assert process.returncode == 130
+
+
+@pytest.mark.e2e
+def test_hot_reload():
+    """
+    Execute a rulebook with hot-reload option,
+    check for first action being triggered,
+    then modify the content of the rulebook,
+    check for the other action being triggered.
+    """
+
+    rulebook = utils.BASE_DATA_PATH / "rulebooks/test_hot_reload.yml"
+    cmd = utils.Command(rulebook=rulebook, hot_reload=True)
+
+    LOGGER.info(f"Running command: {cmd}")
+
+    process = subprocess.Popen(
+        cmd,
+        cwd=utils.BASE_DATA_PATH,
+        text=True,
+        stdout=subprocess.PIPE,
+    )
+
+    fin = open(rulebook, "rt")
+    original_data = fin.read()
+    fin.close()
+    found_rule_1_in_out = False
+    found_rule_2_in_out = False
+
+    start = time.time()
+    while line := process.stdout.readline():
+        if "Rule 1: I matched for value_a" in line:
+            found_rule_1_in_out = True
+            break
+        time.sleep(0.1)
+        if time.time() - start > DEFAULT_CMD_TIMEOUT:
+            process.kill()
+
+    assert found_rule_1_in_out
+
+    data = original_data.replace('- action: "value_a"', '- action: "value_b"')
+    fout1 = open(rulebook, "wt")
+    fout1.write(data)
+    fout1.close()
```

Comment:
It is fine, but there is a better a more pythonic way to do it with a context manager:
```suggestion
    with open(rulebook, "rt") as file:
        original_data = file.write()
```

In that way you ensure the file is properly closed even if there is some error.  

Also you can do it as well in oneline if you use the [pathlib library](https://docs.python.org/3/library/pathlib.html) with the [write_text](https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text) method.

Reply:
This comment uses "open read" I believe is a typo. I will manage this suggestion in separate commit "from my end" not from github.

Reply:
ups, things of the copy/paste. Good catch!